reportssql/treatment_report_functions.sql

```diff
@@ -431,3 +434,37 @@ BEGIN
     RETURN (result );
 END$$ 
 DELIMITER ;
+
+-- patientARVDrugDurationBetween
+
+DROP FUNCTION IF EXISTS patientARVDrugDurationBetween;
+
+DELIMITER $$
+CREATE FUNCTION patientARVDrugDurationBetween(
+    p_patientId INT(11),
+    p_drugDurationMonthsMin INT(11),
+    p_drugDurationMonthsMax INT(11)) RETURNS TINYINT(1)
+    DETERMINISTIC
+BEGIN
+
+DECLARE result TINYINT(1) DEFAULT 0;
+
+    SELECT true INTO result
+    FROM orders o
+    JOIN drug_order do ON do.order_id = o.order_id
+    JOIN concept c ON do.duration_units = c.concept_id AND c.retired = 0
+    JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+    WHERE o.patient_id = p_patientId AND o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND calculateTreatmentEndDate(
```

Comment:
@lauravignoli the duration needs to be checked on ARV satisfying the following conditions:
(1) if the current patient is an old patient: an ARV that was dispensed within the reporting period OR was dispensed before the reporting period but with a treatment that covers the entire reporting period
(2) if the current patient is not an old patient: an ARV that was dispensed within the reporting period

The function `isOldPatient(p_patient_id, p_startDate)` checks if a patient is old or not

Reply:
Thanks, will fix it.